### PR TITLE
MAPI-1342: fixes, upgrades, removing logger errors

### PR DESCRIPTION
**Pull request by agulowaty**
- upgraded manangement-internal and salat libraries to remove scala-compiler eviction
- removed eviction on logback
- ExpirationRequest: missing case added
- ContentApiUpdateActor: webPublicationDate is optional - prevent from dereferencing not defined in log message
- removed snapshot repositories
- upgraded to scala 2.11 and libraries
- set proper scopes to test libraries, upgraded spray version

### CODE REVIEW

**Comment by agulowaty** [comment]
cc @alexduf @pvighi @davidfurey

**Comment by alexduf** [comment]
:+1:

File: src/main/scala/com/gu/mobile/notifications/content/actors/ContentApiUpdateActor.scala [hunk 1]
```diff
@@ -32,7 +30,8 @@ class ContentApiUpdateActor(contentApiClient: ContentApiClient, contentProcessor
         case Success(s) =>
           Metrics.contentApiResponseTime.recordTimeSpent(elapsed)
           s.results.foreach { result =>
-            logger.debug(s"Updated content. id: ${result.id}, title: ${result.webTitle}, date: ${result.webPublicationDate}")
+            val contentPublicationDate = result.webPublicationDateOption.fold("")(d => s", date: $d")
```
Comment (davidfurey) [inline src/main/scala/com/gu/mobile/notifications/content/actors/ContentApiUpdateActor.scala]: I think ```result.webPublicationDateOption.map(d => s", date: $d").getOrElse("")``` would be clearer

File: src/test/scala/com/gu/mobile/notifications/content/actors/MessageSendingActorSpec.scala [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
 import com.gu.util.liveblogs.Block
 import com.gu.util.liveblogs.Blog
 
 class MessageSendingActorSpec(_system: ActorSystem) extends TestKit(_system) with ImplicitSender
-  with WordSpec with MustMatchers with BeforeAndAfterAll {
+  with WordSpecLike with MustMatchers with BeforeAndAfterAll {
 
   def this() = this(ActorSystem("MessageSendingActorSpec"))
 
+  def successfullRight = Future.successful(Right(()))
```
Comment (davidfurey) [inline src/test/scala/com/gu/mobile/notifications/content/actors/MessageSendingActorSpec.scala]: typo: ```successfullRight``` should be ```successfulRight```

**Pull request merged**